### PR TITLE
Dutch auction

### PR DESCRIPTION
**Pull request by ash-burnt**

### CODE REVIEW

File: auction/program/src/lib.rs [hunk 1]
```diff
@@ -11,4 +11,4 @@ pub mod processor;
 pub const PREFIX: &str = "auction";
 pub const EXTENDED: &str = "extended";
 pub const BIDDER_POT_TOKEN: &str = "bidder_pot_token";
-solana_program::declare_id!("auctxRXPeJoc4817jDhf4HbjnhEcr1cCXenosMhK5R8");
+solana_program::declare_id!("NH1kx3qaCqEMABLf3H5Dh3XKzFXd6LgoSmTyZj9p3L8");
```
Comment (justinkook) [inline auction/program/src/lib.rs:14]: GsgdMuQZnMAg4zCkc86JGSgZqRq55jTb8iBf4omiKnND

File: auction/program/tests/helpers.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-use mpl_auction::{
+use metaplex_auction::{
```
Comment (justinkook) [inline auction/program/tests/helpers.rs]: use mpl_auction::{

File: metaplex/program/src/lib.rs [hunk 3]
```diff
@@ -10,4 +10,4 @@ pub mod utils;
 // Export current sdk types for downstream users building with a different sdk version
 pub use solana_program;
 
-solana_program::declare_id!("p1exdMJcjVao65QdewkaZRUnU6VPSXhus9n2GzWfh98");
+solana_program::declare_id!("HaR7dBJRSxypJBs8FxRYBqxSY8k9F4AzEmrDr76AjpNX");
```
Comment (justinkook) [inline metaplex/program/src/lib.rs:13]: D1QcuZQoT3fYhc8c8wqyVkimB18gPGLm3D4hBaZbdwpP

File: auction/program/src/processor.rs [hunk 4]
```diff
@@ -550,6 +595,8 @@ impl BidState {
         minimum: u64,
         instant_sale_price: Option<u64>,
         auction_state: &mut AuctionState,
+        decrease_rate: Option<u64>,
```
Comment (ash-burnt) [inline auction/program/src/processor.rs]: @justinkook `auction_state`, `decrease_rate`, and `decrease_interval` are unused in this function. Any idea what the story is with them?

File: auction/program/tests/lib.rs [hunk 5]
```diff
@@ -1,31 +1,30 @@
...
-    errors::AuctionError,
-    instruction,
-    processor::{
-        process_instruction, AuctionData, AuctionState, Bid, BidState, BidderPot, CancelBidArgs,
-        CreateAuctionArgs, PlaceBidArgs, PriceFloor, StartAuctionArgs, WinnerLimit,
-    },
-    BIDDER_POT_TOKEN, PREFIX,
-};
-use mpl_testing_utils::assert_custom_error;
-use num_traits::FromPrimitive;
```
Comment (justinkook) [inline auction/program/tests/lib.rs]: Update this file to latest metaplex. Should be `mpl_auction`

File: auction/program/src/processor.rs [hunk 6]
```diff
@@ -414,10 +433,10 @@ impl AuctionData {
             gap_val,
             minimum,
             instant_sale_price,
-            &mut self.state,
```
Comment (justinkook) [inline auction/program/src/processor.rs:417]: Were these removed because they wernt being used?

File: auction/program/src/processor.rs [hunk 7]
```diff
@@ -549,7 +591,6 @@ impl BidState {
         gap_tick_size_percentage: Option<u8>,
         minimum: u64,
         instant_sale_price: Option<u64>,
-        auction_state: &mut AuctionState,
```
Comment (justinkook) [inline auction/program/src/processor.rs:552]: Latest auction contract includes this param. Is it not being used?

File: auction/program/src/processor/create_auction.rs [hunk 8]
```diff
@@ -160,13 +215,16 @@ pub fn create_auction(
...
+        decrease_rate: Some(send_decrease_rate),
+        decrease_interval: Some(decline_interval),
+        auction_start_time: None,
     }
     .serialize(&mut *accounts.auction_extended.data.borrow_mut())?;
 
     // Configure Auction.
     AuctionData {
         authority: args.authority,
-        bid_state: bid_state,
```
Comment (justinkook) [inline auction/program/src/processor/create_auction.rs:169]: Latest auction contract includes `bid_state: bid_state,`. I'm guessing this is code cleanup to the same assignment logic

File: auction/program/src/processor.rs (hunk 6 above)
Comment (ash-burnt) [reply auction/program/src/processor.rs:417]: Yes, they were also removed from the function signature

File: auction/program/src/processor/create_auction.rs (hunk 8 above)
Comment (ash-burnt) [reply auction/program/src/processor/create_auction.rs:169]: Yes, it's the preferred way of submitting an arg that has the same name as the var submitted

File: auction/program/src/processor/create_auction.rs [hunk 9]
```diff
@@ -99,11 +100,65 @@ pub fn create_auction(
...
+            2 => BidState::new_dutch(n),
+            _ => BidState::new_english(n),
+        },
         WinnerLimit::Unlimited(_) => BidState::new_open_edition(),
     };
 
+    let mut decline_interval = 0;
+    let lamp = 1000000000;
+    let decline_rate = 2 * lamp;
+    let mut send_decrease_rate: u64 = 0;
```
Comment (justinkook) [inline auction/program/src/processor/create_auction.rs:120]: decreaes rate and decrease interval was going to be custom, using the params passed in but was hard coded instead.

**Pull request merged**